### PR TITLE
Overhaul of DDL syntax variants

### PR DESCRIPTION
**Pull request by nene**
Covering the following:

- CREATE TABLE
- CREATE VIEW
- DROP TABLE
- ALTER TABLE

There are of course more, but these are the ones so far documented in our wiki.

The grammar variations of ALTER TABLE are huge. I only covered the very basic and most widespread syntax:

- ADD COLUMN
- DROP COLUMN
- ALTER COLUMN
- RENAME COLUMN
- RENAME TO
- MODIFY

The tests again don't attempt to cover all the variations in syntax, only the most important bits.

### CODE REVIEW

File: test/postgresql.test.ts [hunk 1]
```diff
@@ -117,4 +124,50 @@ describe('PostgreSqlFormatter', () => {
...
+        foo
+      DROP DEFAULT
+      ;
+
+      ALTER TABLE
+        t
+      ALTER COLUMN
+        foo
+      SET NOT NULL
+      ;
```
Comment (nene) [inline test/postgresql.test.ts:164]: Looks like we have a bug in formatting of semicolons. Won't fix in this PR. Will tackle it separately.

**Comment by inferrinizzard** [comment]
Why DDL this time and not Data Definition like the other 2 PRs ?

**Comment by nene** [comment]
Valid point :) will change.

File: test/bigquery.test.ts [hunk 2]
```diff
@@ -441,56 +394,23 @@ describe('BigQueryFormatter', () => {
...
+          field_2 INT64
+        )
+        OPTIONS(
+          format = 'CSV',
+          uris = ['gs://bucket/path1.csv']
+        )`;
+      const expected = dedent`
+        CREATE EXTERNAL TABLE
+          dataset.CsvTable
+        WITH PARTITION COLUMNS
```
Comment (inferrinizzard) [inline test/bigquery.test.ts:411]: this output seems worse than the input somehow

**Review by inferrinizzard — APPROVED** [review]

File: test/bigquery.test.ts (hunk 2 above)
Comment (nene) [reply test/bigquery.test.ts:411]: I agree.

There's a larger problem with our current formatting style, which wants to format everything as:

```
SOME CLAUSE
  content
ANOTHER CLAUSE
  clause content
```

While it often makes more sense to not create an indented block, but instead place things one the same line. For example `SORT BY` and `LIMIT` are more commonly formatted on the same line:

```sql
SORT BY col1 ASC, col2 DESC
LIMIT 10, 8
```

Similarly `CREATE TABLE` is usually formatted as:

```sql
CREATE TABLE customer (
  id INT NOT NULL,
  name VARCHAR NOT NULL,
  address VARCHAR
);
```

But the formatter currently produces:

```sql
CREATE TABLE
  customer (
    id INT NOT NULL,
    name VARCHAR NOT NULL,
    address VARCHAR
  );
```

...which isn't really how anybody in would format it manually.

We'll need to do some rethinking and reorganization of the current formatting approach.

**Pull request merged**